Highway.h

```diff
@@ -3,14 +3,20 @@
 #include <vector>
 #include <algorithm>
 
+#include "HighwayPatrol.h"
```

Comment:
no need to include this. just forward declare it so you can make it a friend in the class at the end.
- [ ] revise